src/debug.c

```diff
@@ -1031,15 +1031,15 @@ NULL
 
 __attribute__ ((noinline))
 void _serverAssert(const char *estr, const char *file, int line) {
-    bugReportStart();
+    int new_report = bugReportStart();
     serverLog(LL_WARNING,"=== ASSERTION FAILED ===");
     serverLog(LL_WARNING,"==> %s:%d '%s' is not true",file,line,estr);
 
     if (server.crashlog_enabled) {
 #ifdef HAVE_BACKTRACE
         logStackTrace(NULL, 1);
 #endif
-        printCrashReport();
+        if (new_report) printCrashReport();
```

Comment:
maybe add a comment? i.e. we assume we got an assertion during info collection, so we just print the stack trace in this case.

p.s. i'm not sure even that stack trace is needed, since we do print the assertion details, but i certainly agree we wanna stop the recursion.

Reply:
It depends on the specific assertion. The one we ran into at AWS was the refcount on robj, which is not a super useful assertion to hit since it comes from various places.

Reply:
ok, so just add a comment.